src/libraries/System.Private.CoreLib/src/System/Numerics/Matrix4x4.Impl.cs

```diff
@@ -165,106 +165,22 @@ public Vector3 Translation
-                    var brodYz = Vector128.Create(left.Y.Z);
-                    var brodYw = Vector128.Create(left.Y.W);
-                    result.Y = (brodYx * rowX + brodYy * rowY + brodYz * rowZ + brodYw * rowW).AsVector4();
-
-                    var brodZx = Vector128.Create(left.Z.X);
-                    var brodZy = Vector128.Create(left.Z.Y);
-                    var brodZz = Vector128.Create(left.Z.Z);
-                    var brodZw = Vector128.Create(left.Z.W);
-                    result.Z = (brodZx * rowX + brodZy * rowY + brodZz * rowZ + brodZw * rowW).AsVector4();
-
-                    var brodWx = Vector128.Create(left.W.X);
-                    var brodWy = Vector128.Create(left.W.Y);
-                    var brodWz = Vector128.Create(left.W.Z);
-                    var brodWw = Vector128.Create(left.W.W);
-                    result.W = (brodWx * rowX + brodWy * rowY + brodWz * rowZ + brodWw * rowW).AsVector4();
-
-                    return result;
-                }
-                else
-                {
-                    Impl result;
-
-                    // result.X = Transform(left.X, in right);
-                    result.X = right.X * left.X.X;
-                    result.X += right.Y * left.X.Y;
-                    result.X += right.Z * left.X.Z;
-                    result.X += right.W * left.X.W;
-
-                    // result.Y = Transform(left.Y, in right);
-                    result.Y = right.X * left.Y.X;
-                    result.Y += right.Y * left.Y.Y;
-                    result.Y += right.Z * left.Y.Z;
-                    result.Y += right.W * left.Y.W;
-
-                    // result.Z = Transform(left.Z, in right);
-                    result.Z = right.X * left.Z.X;
-                    result.Z += right.Y * left.Z.Y;
-                    result.Z += right.Z * left.Z.Z;
-                    result.Z += right.W * left.Z.W;
-
-                    // result.W = Transform(left.W, in right);
-                    result.W = right.X * left.W.X;
-                    result.W += right.Y * left.W.Y;
-                    result.W += right.Z * left.W.Z;
-                    result.W += right.W * left.W.W;
+                Impl result;
+                result.X = Transform(left.X, in right);
+                result.Y = Transform(left.Y, in right);
+                result.Z = Transform(left.Z, in right);
+                result.W = Transform(left.W, in right);
+                return result;
+            }
 
-                    return result;
-                }
+            [MethodImpl(MethodImplOptions.AggressiveInlining)]
+            private Vector4 Transform(Vector4 vector, in Impl matrix)
+            {
+                var result = matrix.X * vector.X;
+                result += matrix.Y * vector.Y;
+                result += matrix.Z * vector.Z;
+                result += matrix.W * vector.W;
+                return result;
             }
```

Comment:
This is already defined as `Vector4.Transform`, please don't duplicate it unnecessarily